nds-org/workbench-apiserver-python
# feat: scale-down script

## Problems
Some users leave their Workbench applications running long after they stop using them

## Approach
* feat: add a script to scale down userapps that have been running for longer than 10 days

This could be run in the future via `CronJob`

## How to Test
1. Checkout and run this branch locally
2. Adjust `job.yaml` for values of namespace / NAMESPACE / serviceAccountName
3. Create the job on your cluster: `kubectl apply -f job.yaml`
4. Check the logs to see status/progress: `kubectl logs -f job/scaledown -n NAMESPACE`

### TODOs
* implement `FORCE` envvar to force a scale-down (to shut down apps that have been running for less than 10 days)
